src/Microsoft.AspNetCore.SystemWebAdapters/Authentication/IAuthenticationResultFactory.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.Authentication;
+
+/// <summary>
+/// Interface for creating authentication results from HTTP responses returned
+/// by a remote app in remote authentication scenarios.
+/// </summary>
+public interface IAuthenticationResultFactory
```

Comment:
Does this need to be public? Or is it an implementation detail? Do we expect people to need to override this?

Reply:
I think it's unlikely users will override this, so I've made it internal.